lib/validators/liveValidator.ts

```diff
@@ -814,30 +814,42 @@ export class LiveValidator {
     return matchedPaths
   }
 
-  private async getSwaggerPaths(): Promise<string[]> {
+  /**
+   * Filter swagger file from 'this.options.swaggerPaths'. If pathsPattern is empty return all swagger file.
+   * @param pathsPatterns path pattern to match swagger file
+   */
+  public async getSwaggerPaths(pathsPatterns: string[]): Promise<string[]> {
     if (this.options.swaggerPaths.length !== 0) {
       this.logging(
         `Using user provided swagger paths. Total paths count: ${this.options.swaggerPaths.length}`
       )
       return this.options.swaggerPaths
     } else {
       const allJsonsPattern = path.join(this.options.directory, "/specification/**/*.json")
-      const swaggerPathPatterns: string[] = []
-      if (
-        this.options.swaggerPathsPattern === undefined ||
-        this.options.swaggerPathsPattern.length === 0
-      ) {
+      if (pathsPatterns === undefined || pathsPatterns.length === 0) {
         return this.getMatchedPaths(allJsonsPattern)
       } else {
-        this.options.swaggerPathsPattern.map(item => {
-          swaggerPathPatterns.push(path.join(this.options.directory, item))
-        })
-        return this.getMatchedPaths(swaggerPathPatterns)
+        const patterns = pathsPatterns.map(item => path.join(this.options.directory, item))
+        return this.getMatchedPaths(patterns)
       }
     }
   }
 
-  private async getSwaggerInitializer(swaggerPath: string): Promise<void> {
+  public async refreshResourceProviderCache(resourceProviders: string[]): Promise<void> {
+    const pathsPatterns = resourceProviders.map(
+      it => `/specification/**/resource-manager/${it}/**/*.json`
```

Comment:
Could you use a better name to replace 'it'?

Reply:
updated